core/macros/src/lib.rs

```diff
@@ -148,3 +166,14 @@ pub fn enum_trait_object(args: TokenStream, item: TokenStream) -> TokenStream {
 
     out.into()
 }
+
+#[proc_macro_attribute]
+pub fn no_dynamic(_args: TokenStream, item: TokenStream) -> TokenStream {
```

Comment:
as moulins proposed, this is probably not needed if you just strip the attribute in proc-macro